### ISSUE TITLE
TGM Debug / Correctness Check Roadmap

### ISSUE THREAD

**Issue by shenyangHuang** [tgm-team/tgm#133]
Currently there is MRR performance mismatch between `tgm` implementation and TGB example code.  To systematically debug and also to setup a pipeline for debugging in future additions, I propose the following pipeline (part of it can go into integration test as well)

## Questions to answer ##

- [ ] Does the difference exist in training stage or inference stage? 
- [ ] Do we achieve the same training loss / performance for TGN and TGAT within the same number of epochs when compared to TGB implementation and DyGLib implementation? 
- [ ] Does the error occur in 1). neighbor loader, 2). model layer or 3). dataloading (least likely)?

### Unit Test ###

- [ ] add unit test for recency neighbor sampler
- [ ] add unit test for uniform neighbor sampler (also fix error) 
- [ ] for model layers, adding annotation for which attributes it uses from the `DGBatch`. 

### Training debug ###

- measure TGN loss in TGM vs. TGN loss in TGB, benchmark the difference and check all hyperparameters
- measure TGAT loss in TGM vs. TGAT loss in DyGLib, benchmark the difference and check all hyperparameters

### Inference debug ###

- For first edge in each batch of `tgbl-wiki`, measure the MRR from TGM and from TGB, check the difference
- isolate the study the effect of a single forward pass (TGM) vs. many forward pass (TGB) for model performance
